### DISCUSSION TITLE
Homebrew's migration to python 3.9 is unnecessary and too forceful

### DISCUSSION THREAD

**Discussion by kratsg** [Homebrew/discussions#105] [Everyday usage] [answered]
Hi, just a note that this homebrew migration (Homebrew/homebrew-core#62201) completely broke my mac system. I wanted to grab a particular package (asciinema) which has a python 3.8 variation for the python 3.8 i have via homebrew. Getting asciinema remove all my other python (including 3.8) which left all my virtual environments dangling and threw python 3.9 on my system, which many python packages don't support yet... (e.g. jax, tensorflow, etc...) -- so even if it's out, I don't understand why we're forcing the upgrade...

Is there a way to not force users to uproot their entire python? I've got a lot of `dyld: Library not loaded: @executable_path/../.Python` issues every time this happens and my python is entirely from homebrew.

**Comment by kratsg** [comment, marked as answer]
To fix this, I'm doing

```
$ git reset --hard 56a265eb11a92c05efa050a565b5c9af1b97067c
```

inside `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core` and then I'm reinstalling via `HOMEBREW_NO_AUTO_UPDATE=1 brew install python3` to fix everything up.

**Comment by SMillerDev** [comment]
Homebrew is an evergreen package manager and will update to the latest version of dependencies whenever it can.

**Reply by kratsg** [reply]
I get this, but surely there has to be a way to set the default python version uses want to use for each alias? As of now, anything with `/usr/bin/env python3` broke for me, since i now need to update all of those to point at `python3.8`, and then additionally manually add python3.8 to my path. I've gone ahead and removed all python versions from my system and switched to pyenv since I'm running into many more problems this alias switch has created.

**Reply by SMillerDev** [reply]
Not really, because that would mean we need to test all apps with all possible variations of the environment and we simply don't have the maintainer/CI capacity to do that. If you want to have a specific version of tools, use docker or py/rb/whatever-env. If you want to have the latest version, depend on the homebrew version. And if you want to have an outdated broken version use the system version.
